core/node/node_sync/src/batch_transaction_updater/mod.rs

```diff
@@ -0,0 +1,314 @@
+//! Component responsible for updating L1 batch status.
+
+use std::time::Duration;
+
+use tokio::sync::watch;
+use zksync_dal::{ConnectionPool, Core, CoreDal};
+use zksync_eth_client::EthInterface;
+use zksync_health_check::{Health, HealthStatus, HealthUpdater, ReactiveHealthCheck};
+use zksync_types::{
+    aggregated_operations::AggregatedActionType,
+    eth_sender::EthTxFinalityStatus,
+    web3::{BlockId, BlockNumber, TransactionReceipt},
+    Address, H256, U64,
+};
+use zksync_web3_decl::error::EnrichedClientError;
+
+use self::l1_transaction_verifier::L1TransactionVerifier;
+
+mod l1_transaction_verifier;
+
+#[cfg(test)]
+mod tests;
+
+struct SLBlockNumbers {
+    fast_finality: U64,
+    finalized: U64,
+}
```

Comment:
we have `L1BlockNumbers` struct, why not using it?

Reply:
At first I did not want to import from eth_sender, but I think there is space to unify some logic here. I will rework this.

Reply:
I would need to move `L1BlockNumbers` struct to a different crate as we currently don't depend on the `zksync_eth_sender` create and also this struct is a non public module. Seems like a huge mess to reuse this one struct. Unless you see a good place to move it to where we can use it from both creates.

Reply:
You can move it to types